src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (!target.hidden && target instanceof Herbivore) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(target), 1);
```

Comment:
Before using `splice` to remove an animal from the `Animal.alive` array, you should check if the `indexOf` method actually found the animal (i.e., it didn't return -1). This prevents potential bugs where an animal that is not in the array could cause the last animal to be removed accidentally.